processing/src/main/java/org/apache/druid/math/expr/vector/FunctionErrorReportingExprVectorProcessor.java

```diff
@@ -57,7 +58,7 @@ public ExprEvalVector<T> evalVector(Expr.VectorInputBinding bindings)
                           .ofCategory(DruidException.Category.INVALID_INPUT)
                           .build(e, "Function[%s] encountered exception: %s", functionName, e.getMessage());
     }
-    catch (DruidException e) {
+    catch (DruidException | IAE e) {
```

Comment:
I think it might be better to catch the IAE and convert it to a DruidException of type INVALID_INPUT instead, as it contains richer info for sending over the wire.